cpp/src/arrow/compute/kernels/aggregation.cc

```diff
@@ -0,0 +1,46 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/compute/kernels/aggregation.h"
+
+#include "arrow/compute/kernels/monoid.h"
+#include "arrow/status.h"
+
+namespace arrow {
+namespace compute {
+
+Status AggregateUnaryKernel::Call(FunctionContext* ctx, const Datum& input, Datum* out) {
+  switch (input.kind()) {
```

Comment:
this seems like a slightly different pattern then was taken for UnaryArrayKernel (i.e. this pattern wasn't in the kernel, but instead in util-internal.h).  I haven't thought through if a similar pattern should try to be established, have you?

Reply:
I didn't see `util-internal.h`, I'll try to merge both. When I used this piece, I would have preferred to not be concerned with this and hoped that `Datum` provided a ChunkedArray for both cases (wrapping in a single element ChunkedArray when datum is Array).

I'll add this feature.